src/Event.php

```diff
@@ -121,6 +121,13 @@ public function __get($name)
             return $this->getSortDate();
         }
 
+        if ($name === 'source') {
```

Comment:
This change seems unrelated to the OAuth 2 authentication. Could you move all unrelated changes to a separate PR?